src/Native/configure.cmake

```diff
@@ -226,6 +226,39 @@ check_include_files(
     linux/rtnetlink.h
     HAVE_LINUX_RTNETLINK_H)
 
+# getdomainname on OSX takes an 'int' instead of a 'size_t'
+# check if compiling with 'size_t' would cause a warning
+set (PREVIOUS_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+set (CMAKE_REQUIRED_FLAGS "-Weverything")
+check_cxx_source_compiles(
+    "
+    #include <unistd.h>
+	int main() { size_t namelen = 20; char name[namelen]; getdomainname(name, namelen); }
+    "
+    HAVE_GETDOMAINNAME_SIZET
+	FAIL_REGEX "-Wshorten"
+)
+set (CMAKE_REQUIRED_FLAGS ${PREVIOUS_CMAKE_REQUIRED_FLAGS})
```

Comment:
Conversely, we can have:

```cmake
check_prototype_definition(
    getdomainname
    "int getdomainname(char *name, int namelen)"
    "0"
    "unistd.h"
    HAVE_GETDOMAINNAME_WITH_INT_NAMELEN)
```

`HAVE_GETDOMAINNAME_WITH_INT_NAMELEN` is `0` on Linux and `1` on FreeBSD and OS X.